src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -455,19 +455,18 @@ private TypeSymbol BindTupleType(TupleTypeSyntax syntax, DiagnosticBag diagnosti
 
             if (hasExplicitNames)
             {
-                // If the tuple type with names is bound in a declaration
-                // context then we must have the TupleElementNamesAttribute to emit
-                if (syntax.IsTypeInContextWhichNeedsTupleNamesAttribute())
-                {
-                    // Report diagnostics if System.String doesn't exist
-                    this.GetSpecialType(SpecialType.System_String, diagnostics, syntax);
+                // If the tuple type with names is bound we must have the TupleElementNamesAttribute to emit
+                // it is typically there though, if we have ValueTuple at all
+                // and we need System.String as well
```

Comment:
Why are we checking for the existence of the attribute in binding rather than emit?